### PR TITLE
Add new api fields to application and declaration, behind LP_SELF_SERVE feature flag

### PR DESCRIPTION
**Pull request by slawosz**
https://dfedigital.atlassian.net/browse/CPDNPQ-2994
https://dfedigital.atlassian.net/browse/CPDNPQ-2995

### CODE REVIEW

File: spec/serializers/api/application_serializer_spec.rb [hunk 1]
```diff
@@ -256,6 +256,32 @@
...
+        before do
+          Flipper.enable(Feature::LP_SELF_SERVE)
+        end
+
+        it "serializes the `works_as_senco`" do
+          expect(attributes["works_as_senco"]).to eq(application.senco_in_role)
+        end
+
+        it "serializes the `senco_start_date`" do
+          expect(attributes["senco_start_date"]).to eq(application.senco_start_date.to_json)
```
Comment (alkesh) [inline spec/serializers/api/application_serializer_spec.rb:272]: we probably need to test a non-senco application with the feature flag on - to check `attributes["senco_start_date"] == nil`
